### DISCUSSION TITLE
Search; user export features for user management

### DISCUSSION THREAD

**Discussion by ssoone** [npm/feedback#586] [Website]
As an access manager in a large enterprise, I would say NPM is one of the most annoying services to manage users in for me at this moment. I'll just bring out a couple of things below as an example.

* There is no search feature on the members page. I need to go through a bunch of pages to find the user if they're somewhere in the middle. E.g. I have 40 pages of users; I need someone's user who starts with an M which is on page ~26, so I have to click 4 pages at a time to make it there.
* There is no export users feature. Would be very helpful to be able to export all users from the org with their registered emails for user auditing. 
* Unless the user has filled in their first and last name in their profile, it can be a real guessing game who the user is if their username doesn't reflect their real name. Would be perfect to see the registered email address instead. E.g. user Joe Barrel has a username kittycat23 but is registered with joe.barrel@domain.com
* There's no easy way to tell, which teams the users are part of. Being able to see the users profile within the org would be nice.

Cheers.

**Comment by MylesBorins** [comment]
hey @ssoone this is definitely an interesting use case that we should look into improving. At this time we don't have the capacity to kick off work on it, but I can start some conversations to see if this is something we want to support in the future.

Thanks for opening the discussion!

**Comment by RL4422** [comment]
I agree completely with being able to search and export members.  We are currently trying to write scripts to accomplish this.  We add and remove members on a daily basis at Vercel - and this missing functionality is exhausting.  Mostly because it exists in basically every other tool I add and remove users from.  It's the product I do last because it's so painful.

**Comment by UBNT-MikeTaylor** [comment]
Of all the systems we use, npmjs.com is by far the most painful.

* I can't search users or teams, which is important when removing users.  I have a URL hack bookmarked to show me 100 users per page, but I have to manually search by nickname.  On every other system we can search by email address to find users to be deactivated.

* There is no way to see what teams a user is assigned to.  New users say 'my access should be like Joe' and I have to explain to them that I don't have time to open hundreds of teams to see if Joe is there.  I have to tell them to ask a buddy to look at his team lists, then send them to me.

* The invite screen only shows a limited number of teams which doesn't include the one we use most often.  

* The invite screen only allows picking one team, if you are lucky enough to be able to pick it (see above).

* No SSO/SAML/SCIM support

My onboarding process involves inviting the user, giving them an explanation of all the issues with npmjs management interface and closing the ticket.  I ask them to tell me when they have accepted and then go ask someone else on their team for the list of teams they are on.  Then, when the user re-opens the ticket they can give me the information needed to update their account.

My offboarding process involves manually searching nicknames of hundreds of users to try to find a match against the employee's actual first/last name.  

It's pretty primitive, to be honest.
